packages/Calendar/src/MonthView/MonthView.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+import React, { useContext, useCallback } from 'react';
+import { View, TouchableHighlight, StyleSheet, TouchableOpacity, Text } from 'react-native';
+
+import { backgroundColor, textColor } from '../theme';
+import Header from '../common/Header';
+import { getMonthCount, ALL_MONTHS } from '../common/util';
+import CalendarContext from '../common/CalendarContext';
+import DayView from '../DayView';
+import YearView from '../YearView';
+
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    backgroundColor,
+  },
+  dateText: {
+    fontSize: 16,
+    color: textColor,
+    fontWeight: 'bold',
+  },
+});
+
+function renderHeaderTitle(year, setView) {
+  return (
+    <>
+      <TouchableOpacity
+        style={{ paddingHorizontal: 5 }}
+        onPress={() => setView(() => YearView)}
+      >
+        <Text allowFontScaling={false} style={styles.dateText}>
+          {year}
+        </Text>
+      </TouchableOpacity>
+    </>
+  );
+}
+
+function renderAllMonth(year, selectMonth) {
+  const months = [];
+  for (let i = 0; i < 3; i += 1) {
+    const row = [0, 1, 2, 3].map((m) => {
+      const month = m + 4 * i;
+      return (
+        <TouchableHighlight
+          key={m}
+          style={{ flex: 1, alignItems: 'center', paddingVertical: 10 }}
+          onPress={() => selectMonth(year, month)}
+        >
+          <Text>{ALL_MONTHS[month]}</Text>
+        </TouchableHighlight>
+      );
+    });
+    months.push(<View key={i} style={{ flexDirection: 'row' }}>{row}</View>);
+  }
+  return months;
+}
+
+type Props = {
+  setView: React.Node => void,
+}
+
+function MonthView({ setView }: Props) {
+  const [months, setMonths] = useContext(CalendarContext);
+  const handleLayout = useCallback((e) => {
+    const { layout } = e.nativeEvent;
+    setMonths(mnths => getMonthCount(layout, mnths));
+  }, []);
+
+  const shift = useCallback((val) => {
+    setMonths(mnths => mnths.map(d => new Date(
```

Comment:
My instinct tells me this setMonths cannot be used from context.

Reply:
@syaau , but the changes are being reflected

Reply:
When you scroll over to select a month from different year, you will end up on the same page in the CalendarView. Given that you have implemented the cancel button.

Reply:
![Screenshot_1555473059](https://user-images.githubusercontent.com/12614476/56259864-5d976380-60f4-11e9-802c-2b4f3a1c7688.png)
![Screenshot_1555473075](https://user-images.githubusercontent.com/12614476/56259865-5d976380-60f4-11e9-8d0e-6d4c538b965a.png)

the view are different but clear selection still resides, shall I remove clear selection ??